executor/write.go

```diff
@@ -1167,17 +1167,29 @@ func (e *InsertValues) getRows(cols []*table.Column, ignoreErr bool) (rows [][]t
 			return nil, errors.Trace(err)
 		}
 		e.rowCount = uint64(i)
-		rows[i], err = e.getRow(cols, list, ignoreErr)
+		rows[i], err = e.getRow(cols, list, i, ignoreErr)
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
 	}
 	return
 }
 
+func (e *InsertValues) handleErr(col *table.Column, rowIdx int, err error, ignoreErr bool) error {
+	if err == nil {
+		return nil
+	}
+	if types.ErrDataTooLong.Equal(err) {
```

Comment:
Should we extract this as a function, and add some comment for it,
since this is a trade-off.

Reply:
Just one line, no necessary to extract it, I'll add some comments.